### PR TITLE
governance: changing deprecation policy from one release to two.

### LINKED ISSUE #5694: Change deprecation window from 3 to 6 months
While I'm musing on Envoy deprecation process (#5693, #5559) it'd be really helpful if we could change the deprecation window from 3 months / 1 release to 6 months / 2 releases.  For major features it'd give folks time to allocate time to properly qual and test new code paths, and for minor features it'd still allow better customer communication for behavioral changes.

Throwing this out to the community - if no one objects I will make the changes to the tooling, and then we can bike-shed over on #5559 what we want the default deprecation process to be (how long on each phase)

### PR DESCRIPTION
**Pull request by alyssawilk**
This brings deprecation window from between 0-3 months to 3-6 months.  Updating tools / docs accordingly.

Note the deprecation script can be skipped when we cut the next release because of this.
Also fixing up the deprecation script to handle (date)s and (pending)

*Docs Changes*: n/a
*Release Notes*: inline
Fixes #5694

### CODE REVIEW

**Comment by alyssawilk** [comment]
@envoyproxy/maintainers

**Comment by dnoe** [comment]
Yes, I think it's time.

**Comment by alyssawilk** [comment]
I'd be inclined to give complete feature removal a 1y timeline.  I can add that here, or separate out if there may be debate?

File: GOVERNANCE.md [hunk 1]
```diff
@@ -97,7 +97,7 @@
   the same time, also add a new empty "pending" section to the [release
   notes](docs/root/intro/version_history.rst) and to [DEPRECATED.md](DEPRECATED.md) for the
   following version. E.g., "1.7.0 (pending)".
-* Run the deprecate_versions.py script (e.g. `sh tools/deprecate_version/deprecate_version.sh 1.9.0 1.10.0`)
+* Run the deprecate_versions.py script (e.g. `sh tools/deprecate_version/deprecate_version.sh 1.8.0 1.9.0 1.10.0`)
```
Comment (mattklein123) [inline GOVERNANCE.md]: Don't you mean to remove the `1.9.0` in the invocation here?

**Comment by htuch** [comment]
@alyssawilk let's make it a year for now, to preserve the relative proportions, and we can debate whether we need the escape hatch long term in a separate thread.

**Comment by mattklein123** [comment]
> @alyssawilk let's make it a year for now, to preserve the relative proportions, and we can debate whether we need the escape hatch long term in a separate thread.

Can you clarify publicly what is being discussed here?

**Comment by alyssawilk** [comment]
Clarified in docs - bumping the "remove a feature with no replacement" timeline.  I think we still need to be able to remove things entirely (like the move from NPN -> ALPN) but it'd be good to announce farther in advance than we do for simple feature migrations.

Please take a look again - definitely something we want broad support on!

**Comment by alyssawilk** [comment]
Though @htuch I'm not really sure what is covered by that paragraph other than outright removals.  If you have another example I could add it in?  I think 1y for things we're removing entirely is OK as that should be really infrequent but I'm less clear on what else might be affected and would love to clarify.

**Comment by htuch** [comment]
@alyssawilk behavior changes. E.g. if Envoy decides it wants to change the handling of x-foo-bar to translate it to x-f00-B4R, this would be a data plane behavioral change that should also have similar restrictions applied.

**Comment by alyssawilk** [comment]
We don't even guard things like that today :-(

Admittedly I think that ought to change, but I'd prefer we start out with 6 months and scale up if we have to.  Arguably changes like fixing #5554 ought to be flag guarded by this policy but I think we still get enough of them that setting at 1y is going to result in a reaaally messy code base.

Updated docs accordingly.

**Pull request merged**